### PR TITLE
Fix TypeScript warnings

### PR DESCRIPTION
**Pull request by joe-noel-dev**
The main change is enabling strict mode on TypeScript and fixing the warnings.

There are some further improvements to be made in clarifying the lifetime of some of the elements.

### CODE REVIEW

File: scripts/generate.ts [hunk 1]
```diff
@@ -3,10 +3,10 @@ import * as platform from './lib/platform';
 import paths from './lib/paths';
 import {removeDirectory, createDirectory} from './lib/directory';
 
-const skipClean = process.env.SKIP_CLEAN || false;
+const skipClean = process.env['SKIP_CLEAN'] || false;
```
Comment (joe-noel-dev) [inline scripts/generate.ts:6]: See [noPropertyAccessFromIndexSignature](https://www.typescriptlang.org/tsconfig#noPropertyAccessFromIndexSignature)

File: source/ts/app-connection.ts [hunk 2]
```diff
@@ -77,10 +78,6 @@ export class AppConnection extends EventEmitter {
       this.stopServer();
     });
 
-    this.server.on('close', () => {
```
Comment (joe-noel-dev) [inline source/ts/app-connection.ts:80]: Not a particularly useful action. The server shouldn't close without being closed.

File: source/ts/server.ts [hunk 3]
```diff
@@ -2,71 +2,74 @@ import net, {Socket} from 'net';
 import {EventEmitter} from 'events';
 
 export class Server extends EventEmitter {
-  server?: net.Server;
+  listenSocket?: net.Server;
```
Comment (joe-noel-dev) [inline source/ts/server.ts:5]: I thought this is a clearer name, rather than `Server` having a `server`

File: source/ts/server.ts [hunk 4]
```diff
@@ -2,71 +2,74 @@ import net, {Socket} from 'net';
...
 
   constructor() {
     super();
   }
 
   close() {
-    if (this.server) {
-      this.server.close();
-      this.server = null;
+    if (this.listenSocket) {
```
Comment (joe-noel-dev) [inline source/ts/server.ts:13]: Socket will only be cleared on the close event

File: tsconfig.json [hunk 5]
```diff
@@ -1,16 +1,19 @@
...
+    "allowJs": false,
+    "target": "ES2020",
+    "lib": ["ES2020"],
+    "module": "commonjs",
+    "resolveJsonModule": true,
+    "esModuleInterop": true,
+    "sourceMap": true,
+    "declaration": true,
+    "declarationMap": true,
+    "skipLibCheck": true,
```
Comment (joe-noel-dev) [inline tsconfig.json:13]: `skipLibCheck` prevents errors that from `.d.ts` files that assume a DOM

File: source/ts/server.ts (hunk 3 above)
Comment (hmorl) [reply source/ts/server.ts:5]: Screenshot 2023-12-12 at 16 12 21

**Review by hmorl — APPROVED** [review]
Looks good! 🙌

**Review by AbhinavDwarapu — APPROVED** [review]
Nice! 👍

**Pull request merged**